src/Tasks/SystemState.cs

```diff
@@ -573,6 +592,47 @@ private string[] GetDirectories(string path, string pattern)
             return getDirectories(path, pattern);
         }
 
+        /// <summary>
+        /// Cached implementation of GetFiles aimed to verify existence of a file in a directory.
+        /// It does not throw if directory do not exists.
+        /// </summary>
+        /// <param name="path"></param>
+        /// <param name="fileName"></param>
+        /// <returns>file full path or null if file do not exists</returns>
+        private string GetDirectoryFile(string path, string fileName)
+        {
+
+            instanceLocalDirectoryFiles.TryGetValue(path, out Dictionary<string, string> cached);
+            if (cached == null)
+            {
+                string[] files;
+                try
+                {
+                    files = getFiles(path, "*");
+                }
+                catch(DirectoryNotFoundException)
+                {
+                    files = Array.Empty<string>();
+                }
+
+                cached = new Dictionary<string, string>(files.Length, StringComparer.OrdinalIgnoreCase);
+                foreach(var file in files)
```

Comment:
minor:  consider string instead of var for consistency.

Reply:
Code was reworked